### ISSUE TITLE
Bug: [prefer-regexp-exec] Runtime errors on malformed regular expressions

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [typescript-eslint/typescript-eslint#6928]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

Hello friends,

The rule `prefer-regexp-exec` raises a runtime error in case the argument of `String#match` denotes a malformed regular expression. 

For instance, linting

```typescript
'foo'.match('[malformed');
```

gives

```
Oops! Something went wrong! :(

ESLint: 8.38.0

SyntaxError: Invalid regular expression: /[malformed/: Unterminated character class
Occurred while linting /tmp/buggy-prefer-regexp-exec/file.ts:1
Rule: "@typescript-eslint/prefer-regexp-exec"
    at RegExp (<anonymous>)
    at CallExpression[arguments.length=1] > MemberExpression.callee[property.name='match'][computed=false] (/tmp/buggy-prefer-regexp-exec/node_modules/@typescript-eslint/eslint-plugin/dist/rules/prefer-regexp-exec.js:112:36)
    at ruleErrorHandler (/tmp/buggy-prefer-regexp-exec/node_modules/eslint/lib/linter/linter.js:1077:28)
    at /tmp/buggy-prefer-regexp-exec/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/tmp/buggy-prefer-regexp-exec/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/tmp/buggy-prefer-regexp-exec/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/tmp/buggy-prefer-regexp-exec/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/tmp/buggy-prefer-regexp-exec/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at CodePathAnalyzer.enterNode (/tmp/buggy-prefer-regexp-exec/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:795:23)
```

After looking at the rule's implementation, the error is due to this particular line of code:

https://github.com/typescript-eslint/typescript-eslint/blob/f1a0d78ddae418e13ae82f258d951690bb594eaf/packages/eslint-plugin/src/rules/prefer-regexp-exec.ts#L128

It calls the constructor function `RegExp` on the argument of `String#match`. However, the constructor throws a runtime error if the (statically known) regex is malformed.

I would argue that the rule should not break even if it's linting a string literal denoting a malformed regular expression. Instead, it should just keep silent and ignore the visited node. Therefore, a simple fix could be:

```typescript
let regExp: RegExp;
try {
    regExp = RegExp(argumentNode.value);
} catch {
    return;
}
```

Happy to provide a PR if the bug I reported and the fix I suggested make sense.

### Reproduction Repository Link

https://github.com/yassin-kammoun-sonarsource/buggy-prefer-regexp-exec

### Repro Steps

1) clone the repo
2) `npm install`
3) `npx eslint .`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.59.0` |
| `@typescript-eslint/parser`            | `5.59.0` |
| `TypeScript`                           | `5.0.4` |
| `ESLint`                               | `8.38.0` |
| `node`                                 | `18.12.1` |

**Comment by JoshuaKGoldberg** [comment]
Thanks for filing! Great investigation - agreed with your conclusion. Would love a PR! 😍 

By the way, for rule failures reproducible in the online playground, you can just use the smaller template that asks for a playground repro 😄. No need for a whole reproduction repro when this will do: https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=OQMw9mwHQLYIYBcDGALAFMA2vANuATjAKYAmwAlANwBQQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0Wls4DmiAB7F6wxGXRRe0APbRI4MAF8QKoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

**Comment by yassin-kammoun-sonarsource** [comment]
> Thanks for filing! Great investigation - agreed with your conclusion. Would love a PR! 

Will do :)

> By the way, for rule failures reproducible in the online playground, you can just use the smaller template that asks for a playground repro 😄.

How silly of me. I didn't realize I could disable the AST viewer and get the logs. Thanks for the tip!
